packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -518,6 +518,13 @@ function unmount(
             break;
         }
 
+        case VNodeType.Fragment: {
+            if (!lwcRuntimeFlags.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+                unmountVNodes(vnode.children, elm as ParentNode, renderer, false);
+            }
+            break;
+        }
```

Comment:
In the unmount invocations, after the doRemove iteration, the vnodes are manually removed.

Since there isn't a case for VNodeType.Fragment, it does not disconnect them.

This isn't an issue when ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE is true because they will be invoked on renderer.remove.

Reply:
On second though we might want to wrap the `VNodeType.CustomElement` around the `ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE` check too so it doesn't call `disconnectedCallback` twice.

Reply:
@nolanlawson I did some more testing and I added tests to ensure the behavior is the same between LWC 4.x.x and 3.x.x.